hw/ip/otbn/dv/otbnsim/sim/csr.py

```diff
@@ -15,17 +11,32 @@ class CSRFile:
     def __init__(self) -> None:
         self.flags = FlagGroups()
 
+    @staticmethod
+    def _read_field(field_idx: int, field_size: int, val: int) -> int:
+        mask = (1 << field_size) - 1
+        return (val >> (field_size * field_idx)) & mask
+
+    @staticmethod
+    def _write_field(field_idx: int, field_size: int, field_val: int,
+                     old_val: int) -> int:
+        shifted_mask = (
+            (1 << field_size) - field_size) << (field_size * field_idx)
```

Comment:
Is there a typo in the subtraction here? I would have expected `(1 << field_size) - 1`. Also, I think the code might be cleaner as something like this:
```python
    @staticmethod
    def _write_field(field_idx: int, field_size: int, field_val: int,
                     old_val: int) -> int:
        mask = (1 << field_size) - 1
        shift = field_size * field_idx
        return (old_val & ~(mask << shift)) | (field_val << shift)
```

Reply:
Ah yes, I had tested this but I think I may have done some refactoring and failed to test it again so that error slipped through!

I've redone things with your suggested code.

Reply:
Thanks. And I see you also fixed the "deliberate mistake" (ahem) in my definition of `fg`. Oops!